sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -616,8 +616,13 @@ case class UnionExec(children: Seq[SparkPlan]) extends SparkPlan {
     }
   }
 
-  protected override def doExecute(): RDD[InternalRow] =
+  def addExtraInfo: RDD[InternalRow] => RDD[InternalRow] = {
```

Comment:
What's your plan to add this info for all Exec nodes? Add the function into SparkPlan?

Reply:
Different nodes must implement diferent versions of addExtra info because the structure of the nodes and the interesting informatión to show is different.
Of course we could move this function to SparkPlan, with a predefined empty behavior and implement it in the diferent exec nodes that we want to add extra info. Do you think it would be better?